vm/proxyapp/init.go

```diff
@@ -62,6 +62,9 @@ type Config struct {
 	// rpc_server_uri is used to specify plugin endpoint address.
 	// if not specified, we'll connect to the plugin by std[in, out, err].
 	RPCServerURI string `json:"rpc_server_uri"`
+	// server_tls_cert points a TLS certificate used to authenticate the server.
+	// If not provided, the default system certificate pool will be used.
+	ServerTLSCert string `json:"server_tls_cert"`
```

Comment:
The authentication scenarios sorted by complexity level are:
1. No authentication.
2. TLS. Optionally with server certificate pinning.
3. mTLS. It will need client certificate with private key and optional server certificate for the certificate pinning.

This PR adds TLS with server certificate pinning.
Let's think about how to distinguish other scenarios in the future.

server_tls_cert specify optional parameter for [2] or [3]. One more parameter will be needed in the future to distinguish [2] and [3].

Let's add field "security" with allowed values "none" (default if not specified), "tls" and "mtls" for Mutual TLS.
If CA is not specified by "server_tls_cert", [system CAs will be used](https://stackoverflow.com/questions/40051213/where-is-golang-picking-up-root-cas-from)
If needed, we'll add the "client_tls_cert" in the future. It'll be trickier because it'll be a cert with private key and syz-manager deployment will matter. Adding "client_tls_cert" we'll not affect existing configurations.

Reply:
To speed up this PR, you can return "not implemented" for the scenarios that are not covered by the code.